### PR TITLE
Don't use nqp::execname on OpenBSD

### PR DESCRIPTION
**Pull request by Kaiepi**
Fixess one of two parts to #2824

### CODE REVIEW

**Comment by lizmat** [comment]
Thanks!

File: src/main.nqp [hunk 1]
```diff
@@ -2,14 +2,22 @@ use Perl6::Grammar;
...
+my $sep := $config<osname> eq 'MSWin32' ?? '\\' !! '/';
 #?if jvm
 my $execname := nqp::atkey(nqp::jvmgetproperties,'perl6.execname');
 my $exec-dir := nqp::substr($execname, 0, nqp::rindex($execname, $sep));
 #?endif
-#?if !jvm
-my $exec-dir := nqp::substr(nqp::execname(), 0, nqp::rindex(nqp::execname(), $sep));
+#?if moar
+my $exec-dir := $config<osname> eq 'openbsd'
+    ?? $config<prefix> ~ '/bin/perl6-m'
```
Comment (patzim) [inline src/main.nqp]: Isn't the `/perl6-m` part wrong? `$exec-dir` should be the directory the executable resides in, not the executable itself.

Comment (lizmat) [reply src/main.nqp]: /me has no idea

**Comment by timo** [comment]
@Kaiepi can you check if the issue raised in the line comments above is legitimate? it does look like the other parts of the code chop the last bit of the path off with that substr/rindex bit.

**Comment by timo** [comment]
never mind, looks like https://github.com/rakudo/rakudo/commit/e11d0d54e471447efd8d41cc56da0fa130b500ac did this already!

**Comment by patzim** [comment]
@timo The comment is still valid. But since [this](https://github.com/rakudo/rakudo/commit/d3d4b2e8b4fd71cb48e9c83ef5a8b86f486304ce) commit rakudo is not built relocatable on OpenBSD anyways. So one can just revert this change. I'll create a PR this evening.

**Pull request merged**